### DISCUSSION TITLE
question  on the Solar Energy on Building Envelopes tool

### DISCUSSION THREAD

**Discussion by yaron1000** [UMEP-dev/UMEP#488] [Q&A] [answered]
question  on the Solar Energy on Building Envelopes tool
how can I get the  facade wall of buildings data with a code/gis 
i know the info is in the Energyyearwall.txt 
am trying  to know how much Energy gets the facade wall on the polygon in the city
and  have a lot of polygons
thank you very much for the help
and the

**Comment by biglimp** [comment, marked as answer]
We know that the format of Energyyearwall.txt could be improved and we are working on an update. There is no simple way to do this but here is what I would have done:

Run a Python code in the QGIS script editor looking something like this:
`from osgeo import gdal`
`import numpy as np`
`from osgeo.gdalconst import GDT_Float32`

`gdal_data= gdal.Open(string_variable_specifying_filepath_to_the_dem_used_in_SEBE)`
`grid_array = gdal_data.ReadAsArray().astype(np.float)`
`newraster = grid_array * 0.0`
`walldata = np.loadtxt(string_variable_specifying_path_to_Energyyearwall.txt, skiprows = 1)`
`for i in range(0, walldata.shape[0]):`
`   newraster[walldata[i, 0] -1, walldata[i, 1] -1] = np.sum(walldata[i, 2:]`

`# save raster`
`rows = gdal_data.RasterYSize`
`cols = gdal_data.RasterXSize`
`outDs = gdal.GetDriverByName("GTiff").Create(string_variable_specifying_path_to_new_raster, cols, rows, int(1), GDT_Float32)`
`outBand = outDs.GetRasterBand(1)`
`outBand.WriteArray(newraster, 0, 0)`
`outBand.FlushCache()`
`outBand.SetNoDataValue(-9999)`
`outDs.SetGeoTransform(gdal_data.GetGeoTransform())`
`outDs.SetProjection(gdal_data.GetProjection())`

The code is not tested and will probably not work the first time (I always mess up rows and columns etc.) But what I do is to create an empty matrix with the same size as the DSM used and fill those numbers with the sum of radiation on a wall pixel and then save as a raster geotiff. Then you need to find a way to relate wall pixels to a specific building using e.g. a vector dataset and zonal statistics.

**Reply by yaron-michael** [reply]
thank you very much for the help
